vignettes/dettl.Rmd

```diff
@@ -199,7 +166,37 @@ The transformed data can be inspected to ensure it looks as expected.
 print(transformed_data)
 ```
 
-The final stage to implement is the load stage. Note that you can use an automatic load stage. This will append the `transformed_data` to existing tables in the database respecting any foreign key constraints configured via the `dettl.yml`. See section below for details about [automatic load](#dettl_auto_load). To implement this open the `R/load.R` file. An automatically generated implementation will already be there for you, this simply takes each data frame in the transformed data list and appends those rows to the relevant table in the database. If something more complicated is required this can be replaced. Expect that this stage is only really a set of update queries to be run on the database implemented in R.
+### Load {#dettl_auto_load}
+
+By default dettl will use an automatic load step to write data to the database. At its most simple this will just take the named list of `transformed_data` and append each of these data frames to the table in the database with matching name. To use the automatic load step requires defining that the automatic load should be used in the load block of `dettl.yml` e.g.
+
+```yml
+load:
+  automatic: TRUE
+  verification_queries: test_queries
+  test: R/test_load.R
+```
+
+The automatic load will load data respecting primary key constraints which are read automatically from the database. For example if we want to load data where we have `transformed_data` taking the form
+
+```{r echo = FALSE}
+people <- data.frame(c(1,2),
+                     c("Alice", "Bob"),
+                     c(25, 45),
+                     c(175, 187),
+                     stringsAsFactors = FALSE)
+colnames(people) <- c("id", "name", "age", "height")
+jobs <- data.frame(c(1, 2),
+                   c("researcher", "developer"),
+                   stringsAsFactors = FALSE)
+colnames(jobs) <- c("person", "job")
+knitr::kable(people, caption = "people")
+knitr::kable(jobs, caption = "jobs")
+```
+
+where `id` is the primary key of table `people` which is used as a foreign key for the `person` column of table `jobs`. To support linking between these tables in the transformed data use temporary unique ids to associate the new rows of data with each other. The tables themselves may already have entries with these primary keys used so when the automatic load is run the primary keys are updated when a new record is added. If you want to link to an existing entry in the database by its primary id then use that exact id as the field value.
+
```

Comment:
Do you need to be specific here on what "temporary unique ids" to use? Negative numbers? Or just numbers that don't already exist as keys?